native/common/jp_booleantype.cpp

```diff
@@ -46,31 +46,40 @@ class JPConversionAsBoolean : public JPConversion
 {
 public:
 
-	virtual jvalue convert(JPJavaFrame* frame, JPClass* cls, PyObject* pyobj) override
+	virtual JPMatch::Type matches(JPMatch &match, JPClass *cls)
+	{
+		if (!PyLong_CheckExact(match.object)
+				&& !PyIndex_Check(match.object))
+			return match.type = JPMatch::_none;
+		match.conversion = this;
+		return match.type = JPMatch::_implicit;
+	}
+
+	virtual jvalue convert(JPMatch &match) override
 	{
 		jvalue res;
-		res.z = JPPyLong::asLong(pyobj) != 0;
+		jlong v = PyLong_AsLongLong(match.object);
+		if (v == -1)
+			JP_PY_CHECK();
+		res.z = v != 0;
```

Comment:
I am continuing to remove some of the minor utility functions from the Python layer.  When I rewrote the ruby/python layer there was a lot of doubly indirect stuff.  It served a purpose when we were still supporting Python2 and there were actually differences, but now that it is dropped I have been pushing those back as it just makes the code easier to follow.